src/client/Client.cc

```diff
@@ -1100,6 +1102,7 @@ Inode * Client::add_update_inode(InodeStat *st, utime_t from,
     auto p = st->xattrbl.cbegin();
     decode(in->xattrs, p);
     in->xattr_version = st->xattr_version;
+    need_snapdir_attr_refresh = true;
```

Comment:
so even when a user xattr is set or removed on the directory by the MDS, the xattr version changes and this might cause the snapdir_attrs of the directory to be refreshed on the client side. I guess it's okay since it's not expensive to refresh the atts?

Reply:
Yeh, that should be ok.